workflows/genome_annotation.nf

```diff
@@ -0,0 +1,121 @@
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    VALIDATE INPUTS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+def summary_params = NfcoreSchema.paramsSummaryMap(workflow, params)
+
+// Validate input parameters
+WorkflowSeqqc.initialise(params, log)
+
+// TODO nf-core: Add all file path parameters for the pipeline to the list below
+// Check input path parameters to see if they exist
+def checkPathParamList = [  params.multiqc_config, params.fasta ]
+for (param in checkPathParamList) { if (param) { file(param, checkIfExists: true) } }
+
+// Check mandatory parameters
+//if (params.input) { ch_input = file(params.input) } else { exit 1, 'Input samplesheet not specified!' }
+
+/*
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    IMPORT LOCAL MODULES/SUBWORKFLOWS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+//
+// SUBWORKFLOW: Consisting of a mix of local and nf-core/modules
+//
+//include { INPUT_CHECK           } from '../subworkflows/local/input_check'
+include { GENE_ANNOTATION       } from '../subworkflows/local/gene_annotation'
+include { PHAGE                 } from '../subworkflows/local/phage'
+include { PANGENOME_ANALYSIS    } from '../subworkflows/local/pangenome_analysis'
+include { PLASMIDS              } from '../subworkflows/local/plasmids'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    IMPORT NF-CORE MODULES/SUBWORKFLOWS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+//
+// MODULE: Installed directly from nf-core/modules
+//
+include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoftwareversions/main'
+
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    RUN MAIN WORKFLOW
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+
+
+workflow ANNOTATION {
+    // Declare input and output channels
+
+    main:
+    ch_versions = Channel.empty()
+    genome = file(params.assembled_genome, checkIfExists: true)
+    ch_genome = [ [ id:genome.getBaseName() ], genome ]
+    print(ch_genome)
+
+    // Eventually this will replace the above line for taking input
+    // SUBWORKFLOW: Read in samplesheet, validate and stage input files
+
+    /*INPUT_CHECK (
+        ch_input
+    )
+    ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
+    */
+
+    if (!params.skip_gene_annotation) {
+        // Annotate genomes using Prokka and bakta
+        GENE_ANNOTATION(ch_genome)
+        ch_versions = ch_versions.mix(GENE_ANNOTATION.out.versions)
+
+        if (!params.skip_pangenome_analysis) {
+        // Run pangenome analysis
+        PANGENOME_ANALYSIS()
```

Comment:
The pan-genome analysis workflow should accept either `GENE_ANNOTATION.out.bakta_gff` or `GENE_ANNOTATION.out.prokka_gff` as input.

Reply:
The GFF files will need to be collected and emitted as a list of files

Reply:
updated with [4e713b5](https://github.com/cidgoh/nf-seqqc/pull/58/commits/4e713b57a06a7813e92a7bf3e5328a6ecc6666be) & [d4add19](https://github.com/cidgoh/nf-seqqc/pull/58/commits/d4add191e949961678f503285af315f16f79bc29)